modules/service/src/test/scala/ch/epfl/bluebrain/nexus/iam/service/routes/AuthRoutesSpec.scala

```diff
@@ -0,0 +1,101 @@
+package ch.epfl.bluebrain.nexus.iam.service.routes
+
+
+import java.util.UUID
+
+import akka.http.scaladsl.model.Uri.Query
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.{Authorization, Location, OAuth2BearerToken}
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient.UntypedHttpClient
+import ch.epfl.bluebrain.nexus.iam.service.config.AppConfig.OidcConfig
+import org.mockito.Mockito
+import org.mockito.Mockito._
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.{BeforeAndAfter, Matchers, WordSpecLike}
+
+import scala.concurrent.Future
+
+class AuthRoutesSpec  extends WordSpecLike
```

Comment:
Needed to add test cases for failures I guess. What happen when those endpoints you are forwarding in the routes don't respond?
What happens when a client hits the path /oauth2/userinfo without the header of type Authorization ?

Mostly all of it is handled right now by Akka HTTP. However we probably need to add our own RejectionHandling on it.